test/controllers/supplementary/DeclarationTypePageControllerSpec.scala

```diff
@@ -0,0 +1,178 @@
+    }
+
+    "display radio button with question text for additional declaration type" in {
+      withCaching[DeclarationType](None, DeclarationType.formId)
+
+      val result = route(app, getRequest(uri)).get
+
+      contentAsString(result) must include(
+        messages("supplementary.declarationTypePage.inputText.additionalDeclarationType"))
+      contentAsString(result) must include(
+        messages("supplementary.declarationTypePage.inputText.additionalDeclarationType.hint"))
+    }
+
+    "display \"Save and continue\" button" in {
+      withCaching[DeclarationType](None, DeclarationType.formId)
+
+      val result = route(app, getRequest(uri)).get
+
+      contentAsString(result) must include(messages("site.save_and_continue"))
+      contentAsString(result) must include("button id=\"submit\" class=\"button\"")
+    }
+
+    "not populate the form fields if cache is empty" in {
+      withCaching[DeclarationType](None, DeclarationType.formId)
+
+      val result = route(app, getRequest(uri)).get
+
+      contentAsString(result) mustNot include("checked=\"checked\"")
+    }
+
+    "populate the form fields with data from cache" in {
+      withCaching[DeclarationType](Some(DeclarationType("CO", "Y")), DeclarationType.formId)
+
+      val result = route(app, getRequest(uri)).get
+
+      contentAsString(result) must include("checked=\"checked\"")
+    }
+  }
+
+
+  "DeclarationTypePageController on submitDeclarationType" should {
+
+    "display the form page with error" when {
+      "no value provided for declaration type" in {
+        withCaching[DeclarationType](None)
+
+        val formWithoutDeclarationType = JsObject(Map("additionalDeclarationType" -> JsString("Y")))
+        val result = route(app, postRequest(uri, formWithoutDeclarationType)).get
+
+        contentAsString(result) must include("Please enter a value")
+      }
+
+      "no value provided for additional declaration type" in {
+        withCaching[DeclarationType](None)
+
+        val formWithoutAdditionalDeclarationType = JsObject(Map("declarationType" -> JsString("EX")))
+        val result = route(app, postRequest(uri, formWithoutAdditionalDeclarationType)).get
+
+        contentAsString(result) must include("Please enter a value")
+      }
+    }
+
+    "save the data to the cache" in {
+      reset(mockCustomsCacheService)
+      withCaching[DeclarationType](None)
+
+      val validForm = ExportsTestData.correctDeclarationType
+      route(app, postRequest(uri, validForm)).get.futureValue
+
+      verify(mockCustomsCacheService)
+        .cache[DeclarationType](any(), ArgumentMatchers.eq(DeclarationType.formId), any())(any(), any(), any())
+    }
+
+    pending
+    "return 303 code" in {
+      withCaching[DeclarationType](None)
+
+      val validForm = ExportsTestData.correctDeclarationType
+      val result = route(app, postRequest(uri, validForm)).get
+
+      status(result) must be(SEE_OTHER)
+    }
+
+    pending
```

Comment:
Please remove pending from tests

Reply:
Tests are based on acceptance criteria, but one of them is to redirect to consignment-reference page, which will be implemented in the next sprint. This is why I decided to write tests now and mark them as pending.